src/main/java/org/example/controllers/ApplicationController.java

```diff
@@ -26,14 +26,20 @@ public ApplicationController(final ApplicationService applicationService) {
 
     @POST
     @Path("/apply-for-role")
-    @RolesAllowed({UserRole.ADMIN, UserRole.USER})
+    @RolesAllowed(UserRole.USER)
     public Response apply(final ApplicationRequest applicationRequest) {
         try {
-            applicationService.createApplication(
+            boolean valid = applicationService.createApplication(
```

Comment:
I would prefer if you used exceptions to determine which code to return rather than using a boolean here